python/cudf/bindings/binops.pyx

```diff
@@ -142,3 +161,32 @@ def apply_op(lhs, rhs, out, op):
     free(c_out)
 
     return nullct
+  
+def apply_op_udf(lhs, rhs, out, ptx):
```

Comment:
On the python side, somewhere, we should probably provide a higher-level API.  `def apply_op_udf(las, rhs, out, func)`, where `func` is a python function with `@jit` decorator. `apply_op_udf` would determine the `type_signature` (easy for binops?) and do the compilation and ptr inspection, then call this function:

```
func.compile(type_signature)
ptx = add.inspect_ptx(type_signature)
ptx_code = ptx.decode('utf-8')
apply_op_udf(las, rhs, out, ptx_code)
```

@kkraus14 should comment on where and how best to expose this.

Reply:
@harrism Yes, we'll figure out how to expose this at a higher level in a follow up PR, we should just keep this at the libcudf / Cython level for now to continue development and figure out the API.